frontend/src/app/components/auth/auth.service.ts

```diff
@@ -155,7 +155,7 @@ export class AuthService {
   }
 
   isAuthenticated(): boolean {
-    return this._cookie.get('token') !== null;
+    return this._cookie.check('token') && this._cookie.get('token') !== null;
```

Comment:
**Suggestion:** The check for the existence of the 'token' cookie is redundant. The `get` method will return `null` if the cookie does not exist, so checking with `check` method is not necessary. [enhancement]
```suggestion
    isAuthenticated(): boolean {
        return this._cookie.get('token') !== null;
    }
```